### DISCUSSION TITLE
Complete URL

### DISCUSSION THREAD

**Discussion by ben-sec** [Security-Onion-Solutions/securityonion#9146] [Unsupported Versions] [answered]
Hi!

I have an alert for suspicious ipfs.io traffic and I'm wondering if I can find the complete URL/URI that the user has accessed somewhere in SO?

Cheers, Ben

**Comment by Jaap79** [comment]
I might be jumping the queue here, but I think that'd be difficult to accomplish based on the information you provided. Depends on how the URL was accessed too.. (was it forwarded internally by the receiving server for instance?) and do you solely trust suricata (your question suggests an alert from a suricata rule) or do the endpoints have Wazuh agents running for instance? Also, do you ingest firewall logs and what do they tell you?

**Reply by ben-sec** [reply]
I have Suricata alerts, Zeek (Connection, DNS and SSL) logs and also Sysmon logs, but I can only find evidences that the DNS name resolution of ipfs.io triggered the alarm. Maybe the user visited just this URL? My ingested Firewall logs do only show the visited destination ip.

**Reply by Jaap79** [reply]
Would make sense that the resolution alone would trigger this.. a subpage doesn’t make the DNS resolving any different.. but I wouldn’t know exactly how to figure out the full URL either. Unless something was downloaded from the website.. (in which case zeek would have known about it too). 

Maybe you can Hunt for favicon or .html files or something, which might indicate where they came from. Or grab the PCAP of the web request and see if something pops up in there?

**Comment by dougburks** [comment]
> I have Suricata alerts, Zeek (Connection, DNS and SSL) logs and also Sysmon logs, but I can only find evidences that the DNS name resolution of ipfs.io triggered the alarm. Maybe the user visited just this URL? My ingested Firewall logs do only show the visited destination ip.

Depending on your Sysmon configuration, it may have logged the activity and our new Sysmon dashboards might come in handy:
https://blog.securityonion.net/2022/10/security-onion-23180-now-available.html

**Comment by ben-sec** [comment, marked as answer]
Found it: If you click on a link in a phishing mail, Sysmon creates a "process creation" event for your default browser and you will find the full URL in the process.parent.command_line field!
